### ISSUE TITLE
use_nested_configs: true ignores rule in "main" .swiftlint.yml

### ISSUE THREAD

**Issue by derpoliuk** [realm/SwiftLint#675]
I'm using SwiftLint 0.10.0

In my project I have 2 `.swiftlint.yml`: in `MyProject/` and in `MyProject/MyProjectTests/` folders.

```
# MyProject/.swiftlint.yml
line_length: 110
excluded:
 - Carthage
use_nested_configs: true
```

```
# MyProject/MyProjectTests/.swiftlint.yml
disabled_rules:
 - force_cast
 - force_try
```

And SwiftLint generates warnings in whole project, ignoring rule from first `.swiftlint.yml`:

```
Line Length Violation: Line should be 100 characters or less: currently 101 characters (line_length)
```

**Comment by jpsim** [comment]
We deprecated support for `use_nested_configs` in #527, so the value you're setting for that key in the config file is ignored.

Now nested configurations are always considered when present, but that still doesn't explain the behavior you're seeing. I'll try seeing if I can reproduce this.

**Comment by derpoliuk** [comment]
@jpsim thanks for quick response, didn't know that `use_nested_configs` is deprecated.

**Comment by jpsim** [comment]
I can't reproduce this:

```shell
$ tree -a
.
├── .swiftlint.yml
├── test.swift
└── tests
    └── .swiftlint.yml
$ cat test.swift 
//////////////////////////////////////////////////////////////////////////////////////////////////////////////
$ cat .swiftlint.yml 
line_length: 110
$ cat tests/.swiftlint.yml 
disabled_rules:
 - force_cast
 - force_try
$ swiftlint
Loading configuration from '.swiftlint.yml'
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
Done linting! Found 0 violations, 0 serious in 1 file.
$ swiftlint version
0.10.0
```

The `test.swift` file has 110 `/` characters and the configuration from the root `.swiftlint.yml` file is applied so no violations are triggered.

What are you doing differently? Did I miss a step?

**Comment by jpsim** [comment]
> didn't know that use_nested_configs is deprecated.

That's odd, since SwiftLint logs the following message when you include that key in a config file:

> Support for 'use_nested_configs' has been deprecated and its value is now ignored. Nested configuration files are now always considered.

Are you sure you're using SwiftLint 0.10.0?

**Comment by derpoliuk** [comment]
> Are you sure you're using SwiftLint 0.10.0?

Yes, `swiftlint version` prints `0.10.0`.

> That's odd, since SwiftLint logs the following message when you include that key in a config file:

I didn't run it from command line, I'm using Xcode Run Script Phase for that. 

> What are you doing differently? Did I miss a step?

Finally I figured out what the difference: swiftlint generates this warnings for files in `MyProjectTests` folder. I didn't see it when I submitted this issue. Is it expected behavior with nested configs?

**Comment by jpsim** [comment]
Yes, nested configurations *replace* their parents. There's room for more intelligent merging of configurations, which we have commented in the code ([link](https://github.com/realm/SwiftLint/blob/0.10.0/Source/SwiftLintFramework/Models/Configuration.swift#L220-L225)) but I can't find an issue that's currently tracking that.

I've filed #676 to track that improvement. If you're interested in contributing to SwiftLint, that'd be a great place to start!

Feel free to close this issue if you feel I've sufficiently addressed your questions 😄
